src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.AddProperty.cs

```diff
@@ -51,11 +51,13 @@ private JsonPropertyInfo AddProperty(Type propertyType, PropertyInfo propertyInf
         internal JsonPropertyInfo CreateProperty(Type declaredPropertyType, Type runtimePropertyType, PropertyInfo propertyInfo, Type parentClassType, JsonSerializerOptions options)
         {
             Type collectionElementType = null;
-            ClassType propertyClassType = GetClassType(runtimePropertyType);
-            if (propertyClassType == ClassType.Enumerable || propertyClassType == ClassType.Dictionary)
+            switch (GetClassType(runtimePropertyType))
             {
-                collectionElementType = GetElementType(runtimePropertyType);
-                // todo: if collectionElementType is object, create loosely-typed collection (JsonArray).
```

Comment:
Is this todo no longer relevant?

Reply:
Yep. This change fixes this. We put  `JsonElement`s into object collections.